src/Features/Core/Portable/Snippets/SnippetData.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    /// <summary>
+    /// Stores only the data needed for the creation of a CompletionItem.
+    /// Avoids using the Snippet and creating a TextChange/finding cursor
+    /// position before we know it was the selected CompletionItem.
+    /// </summary>
+    internal struct SnippetData
```

Comment:
this type seems entirely unnecessary.

Reply:
maybe call this SNippetInfo?  Data is very vague and doesn't really convey how it is different from Snippet.  Whereas 'Info' indicates into information abotut eh snippet without being what the actual snippet is.